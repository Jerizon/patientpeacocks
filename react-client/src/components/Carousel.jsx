import React from 'react';
import Button from 'react-bootstrap/lib/Button';
import ButtonToolbar from 'react-bootstrap/lib/ButtonToolbar'
import Carousel from 'react-bootstrap/lib/Carousel'

class CarouselPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      direction: null
    }
    this.handleSelect.bind(this);
  }

  handleSelect(selectedIndex, e) {
    var context = this;
    context.setState({
      index: selectedIndex,
      direction: e.direction
    });
  }

  render() {
    return (
      <Carousel className = 'carousel' activeIndex={this.state.index} direction={this.state.direction} onSelect={this.handleSelect.bind(this)}>
        <Carousel.Item>
          <img width={500} height={300} alt="500x300" src="https://cdn.pixabay.com/photo/2014/03/05/19/23/dog-280332_960_720.jpg" />
          <Carousel.Caption>
            <h3>Puppy</h3>
            <p>Cute puppy</p>
          </Carousel.Caption>
        </Carousel.Item>
        <Carousel.Item>
          <img width={500} height={300} alt="500x300" src="http://7r7m646bco73w5dhw1kfab26.wpengine.netdna-cdn.com/wp-content/uploads/2015/08/176.jpg" />
          <Carousel.Caption>
            <h3>2nd puppy</h3>
            <p>hi pupppy</p>
          </Carousel.Caption>
        </Carousel.Item>
        <Carousel.Item>
          <img width={500} height={300} alt="500x300" src="https://www.aspca.org/sites/default/files/adoptable-dogs-your-local-shelter.jpg" />
          <Carousel.Caption>
            <h3>3rd puppy</h3>
            <p>another puppy</p>
          </Carousel.Caption>
        </Carousel.Item>
      </Carousel>
    );
  }
}
export default CarouselPage;