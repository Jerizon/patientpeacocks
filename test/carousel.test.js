import React from 'react';
import ReactDOM from 'react-dom';
import Carousel from '../react-client/src/components/Carousel.jsx';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Component: App', () => {
  const items = ['Learn react', 'rest', 'go out'];

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <Carousel/>
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});