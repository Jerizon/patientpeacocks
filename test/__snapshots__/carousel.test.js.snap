// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match its empty snapshot 1`] = `
<div
  className="carousel carousel slide"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      onClick={[Function]}
    />
     
    <li
      className={null}
      onClick={[Function]}
    />
     
    <li
      className={null}
      onClick={[Function]}
    />
     
  </ol>
  <div
    className="carousel-inner"
  >
    <div
      className="item active"
    >
      <img
        alt="2000x2000"
        height={2000}
        src="https://www.readersintheknow.com/bundles/ritkhome/images/party.jpg"
        width={2000}
      />
      <div
        className="carousel-caption"
      />
    </div>
    <div
      className="item"
    >
      <img
        alt="2000x2000"
        height={2000}
        src="http://media.gettyimages.com/photos/nothing-beats-live-music-picture-id502088147?s=612x612"
        width={2000}
      />
      <div
        className="carousel-caption"
      />
    </div>
    <div
      className="item"
    >
      <img
        alt="300x1000"
        height={0}
        src="http://www.essentialibiza.com/assets/img/ibiza-boat-party-tickets-images/oceanbeat-ibiza-boat-party-show-with-after-party-in-bora-bora-ibiza-1.jpg"
        width={1500}
      />
      <div
        className="carousel-caption"
      />
    </div>
  </div>
  <a
    className="carousel-control left"
    href="#"
    onClick={[Function]}
    role="button"
  >
    <span
      className="glyphicon glyphicon-chevron-left"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control right"
    href="#"
    onClick={[Function]}
    role="button"
  >
    <span
      className="glyphicon glyphicon-chevron-right"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
